### DISCUSSION TITLE
Step definition not found

### DISCUSSION THREAD

**Discussion by shoevancar** [CXwudi/comp5903-easy-cucumber#22] [Q&A] [answered]
I have placed the following two statements as described in the documentation.

Path myFeatureFile = Paths.get("src\\test\\resources\\features\\AcceptanceTestPart1Scoring.feature");
JFeature jFeature = EasyCucumber.build(myFeatureFile, AcceptancePart1ScoringTest.class);

Although, I have made the class and the methods as public and the annotations according to the feature files. I am getting the following error -

_scs.comp5903.cucumber.model.exception.EasyCucumberException: EZCU013: Step definition not found for: GivenStep(stepString=The game has been initialized)._

**Comment by CXwudi** [comment, marked as answer]
Hi, thanks for reporting the issue, would you mind providing a minimal reproducible sample project?

**Reply by shoevancar** [reply]
Hello, thank you for help. Please find a minimal sample project.
https://github.com/shoevancar/easy-cucumber-test

**Reply by CXwudi** [reply]
Thanks for providing the minimal sample project. 

It's just a small mistake, you can't use official Cucumber's annotaion like `@When`, `@Then`, but the annotation came from this tool, like `@JGivenStep`, `@JWhenStep` and etc.

Let me know if this fixes your issue

**Reply by shoevancar** [reply]
Thank you. I have changed the annotations in the sample project. It worked.
Although, it is running successfully, I can see few lines of error/warning. Are they okay?

_SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details._

**Reply by CXwudi** [reply]
They are fine to leave it. 

The warning is there because the tool has added the slf4j logging api, but it requires a slf4j logging implementation, like log4j2 for example. 

I could have added a slf4j implementation into this tool but then it will start to produce a bunch of logs flushing users terminal. So I decided to delete the implementation dependency and let users choose to include one or not.
